packages/core/src/components/context-menu/contextMenu.tsx

```diff
@@ -65,16 +65,15 @@ class ContextMenu extends AbstractComponent<{}, IContextMenuState> {
     }
 
     public hide() {
-        const { onClose } = this.state;
-        this.setState({ isOpen: false, onClose: null });
-        safeInvoke(onClose);
+        safeInvoke(this.state.onClose);
+        this.setState({ isOpen: false, onClose: undefined });
     }
 
     private cancelContextMenu = (e: React.SyntheticEvent<HTMLDivElement>) => e.preventDefault();
 
     private handleBackdropContextMenu = (e: React.MouseEvent<HTMLDivElement>) => {
-        // HACKHACK: React function to remove from the event pool (not sure why it's not in typings #66)
-        (e as any).persist();
+        // HACKHACK: React function to remove from the event pool
```

Comment:
actually this is not a HACKHACK, it's expected usage if you want to use an event in a callback. can you remove the HACKHACK keyword but leave the rest of comment?